File: leetcode-python-go/2335. 装满杯子需要的最短总时长/main.py
from typing import List

amount = [5,4,4]

def fillCups(amount: List[int]) -> int:
    amount.sort()
    if amount[2] > amount[1] + amount[0]:
        return amount[2]
    return (sum(amount) + 1) // 2

print(fillCups(amount))

File: leetcode-python-go/2335. 装满杯子需要的最短总时长/main.go
package main

import (
	"fmt"
	"sort"
)

func fillCups(amount []int) int {
	sort.Ints(amount)
	if amount[2] > amount[1]+amount[0] {
		return amount[2]
	}
	return (amount[0] + amount[1] + amount[2] + 1) / 2
}

func main() {
	amount := []int{5, 4, 4}
	results := fillCups(amount)
	fmt.Println(results)
}
